packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -1055,13 +1057,32 @@ class _RenderDecoration extends RenderBox {
       - topHeight
       - contentPadding.bottom;
     final double alignableHeight = fixAboveInput + inputHeight + fixBelowInput;
-    // When outline aligned, the baseline is vertically centered by default, and
-    // outlinePadding is used to account for the presence of the border and
-    // floating label.
-    final double outlinePadding = _isOutlineAligned ? 10.0 : 0;
-    final double textAlignVerticalOffset = (maxContentHeight - alignableHeight - outlinePadding) * textAlignVerticalFactor;
+    final double maxVerticalOffset = maxContentHeight - alignableHeight;
+    final double textAlignVerticalOffset = maxVerticalOffset * textAlignVerticalFactor;
     final double inputBaseline = topInputBaseline + textAlignVerticalOffset;
 
+    // The three main alignments for the baseline when an outline is present are
+    //
+    //  * top (-1.0): topmost point considering padding.
+    //  * center (0.0): the absolute center of the input ignoring padding but
+    //      accommodating the border and floating label.
+    //  * bottom (1.0): bottommost point considering padding.
+    //
+    // That means that if the padding is uneven, center is not the exact
+    // midpoint of top and bottom. To account for this, the points are
+    // interpolated between independently.
```

Comment:
the above center and below center alignments are interpolated independently

Reply:
Thanks, I was struggling to not say "we" or "I" in this sentence haha.